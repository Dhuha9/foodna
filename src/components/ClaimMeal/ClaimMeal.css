@import url('https://fonts.googleapis.com/css2?family=Roboto&display=swap');

* {
    box-sizing: border-box;
}

.ClaimMeal{
    display: flex;
    flex-direction: row-reverse;
    justify-content:space-evenly;
    align-items:center;
    background-color: #F2F2F2;
    min-height: 100vh;
    color: #000;
    font-family: 'Roboto', sans-serif;
}


.meal{
    font-family: 'Roboto', sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 20px;
    text-align: left;
    padding-left: 50px;
    padding-right: 50px;
    width:40%
}

.image{
    width: 400px;
    box-shadow: 50px 50px #fff;
    margin:50px
}

h4{
    font-family: 'Roboto', sans-serif;
    font-style: normal;
    font-weight: bold;
    font-size: 24px;
    text-align: left;
    margin-bottom: 0px;
    margin-top: 20px;
}

p.location{
    font-family:'Roboto', sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 16px;
    color: rgba(0, 0, 0, 0.5);
    margin-top: 5px;
}

p.quantity{
    font-family:'Roboto', sans-serif;
    font-style: normal;
    font-weight: normal;
    font-size: 18px;
    color:#000;
    margin-top: 30px;
}


.form{
    display: flex;
    flex-direction: column;
    margin-top:30px;
    width:400px;
}

.formStyle{
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    font-family:'Roboto', sans-serif;
    font-style: normal;
    font-weight: normal;
}

sup{
    color: rgb(255, 0, 0)
}

.input{
    background: #FFFFFF;
    border: 1px solid #545D69;
    box-sizing: border-box;
    border-radius: 8px;
    align-self: stretch;
    flex-grow: 0;
    margin: 8px 0px;
    height: 44px;

}

input::after{
    background:#fff;
}

input:focus, textarea:focus, select:focus{
    outline: none;
}

.Submit{
    margin-top:20px;
    background: #03B664;
    border-radius: 8px;
    padding: 14px 18px;
    border: none;
    color: #fff

}

.Submit:hover{
    background: #FFB701
}




@media only screen and (max-width: 1150px) {
    /* For Tablet screens */

    .ClaimMeal{
        display: block;
    }

    .meal{
        width: 100%;
        text-align: center;
    }

    .form{
        width: 70%;
        margin:auto;
        
    }

    h4{
        text-align: center;
    }

}

@media only screen and (max-width: 570px) {
    /* For Mobile screens */
    .image{
        width: 200px;
        box-shadow: 30px 30px #fff;
    }
}