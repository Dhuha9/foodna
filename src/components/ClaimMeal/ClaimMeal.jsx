// /* eslint react/prop-types: 0 */
// // eslint-disable-next-line react/jsx-no-bind
// import React, { Component } from 'react';
// import '../../meals.css'
// import { Button } from 'react-bootstrap';

// export default class MealCard extends Component {
//   render() {
//     return (
//       <div className="card">
//         <div className="image-section">
//           <img src={this.props.img} className="card-img" alt="food" />
//         </div>
//         <div className="card-body">
//           <h5 className="card-title">{this.props.title}</h5>
//           <span className="card-text">{this.props.organization}</span>
//           <Button className="Button">Get This Meal</Button>
//         </div>
//       </div>
//     );
//   }
// }

import React, { Component } from 'react'
import './ClaimMeal.css'
import firebase from '../../firebase'


export default class ClaimMeal extends Component {

  constructor(props) {
    super(props);
    this.state = {
      name: '',
      email:'',
      phone: '',
      organization: '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    this.setState({
      [name]: value
    });
  }

  handleSubmit(event) {
    console.log(this.state)

    event.preventDefault();

    const MealOrder = this.state;
    firebase.firestore().collection('MealOrder').add(MealOrder);
    

    this.setState({
      name: '',
      email:'',
      phone: '',
      organization: '',
    });
    
  }

  render() {
    return (
      <div className="ClaimMeal">
          <div className="meal">
              <img className="image" src="pizza.jpg" alt="" />
              <h4>Chicken Fajita Dishes</h4>
              <p className="location">Fried Chicken  Zayoona</p>
              <p className="quantity">Quantity : 5 Dishes</p>
              <p>Description : Lorem ipsum dolor sit amet consectetur adipisicing elit. Veritatis itaque, mollitia animi sit, commodi corrupti vel harum quod magnam cupiditate tempore illo saepe libero. Incidunt earum a doloribus minima eum.</p>
              
          </div>
          <div className="form">
              <h4 style ={{marginBottom:30}}>Get This Meal</h4>
                <form onSubmit={this.handleSubmit} className="formStyle">
                  <label>Name <sup>*</sup></label>
                  <input name="name" type="text" value={this.state.name} onChange={this.handleChange} className='input'/>
                  <label>Email <sup>*</sup></label>
                  <input name="email" type="text" value={this.state.email} onChange={this.handleChange} className='input'/>
                  <label>Phone <sup>*</sup></label>
                  <input name="phone" type="text" value={this.state.phone} onChange={this.handleChange} className='input'/>
                  <label>Organization <sup>*</sup></label>
                  <input name="organization" type="text" value={this.state.organization} onChange={this.handleChange} className='input'/>
                  <input type="submit" value="Get This Meal" className="Submit" />
                </form>
            </div>
      </div>
  )
  }
}
