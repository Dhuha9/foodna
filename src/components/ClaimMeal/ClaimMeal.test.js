import React from 'react';
import renderer from 'react-test-renderer';
import ClaimMeal from './ClaimMeal';

it ("renders when there is no Claim Meal component", () => {
    const tree = renderer.create(
        <ClaimMeal />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});