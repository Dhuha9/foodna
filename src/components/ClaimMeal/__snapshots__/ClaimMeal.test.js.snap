// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is no Claim Meal component 1`] = `
<div
  className="ClaimMeal"
>
  <div
    className="meal"
  >
    <img
      alt=""
      className="image"
    />
    <h4 />
    <p
      className="location"
    />
    <p
      className="quantity"
    >
      Quantity : 
    </p>
    <p>
      Description : 
    </p>
  </div>
  <div
    className="form"
  >
    <h4
      style={
        Object {
          "marginBottom": 30,
        }
      }
    >
      Get This Meal
    </h4>
    <form
      className="formStyle"
      onSubmit={[Function]}
    >
      <label>
        Name 
        <sup>
          *
        </sup>
      </label>
      <input
        className="input"
        name="name"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label>
        Email 
        <sup>
          *
        </sup>
      </label>
      <input
        className="input"
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label>
        Phone 
        <sup>
          *
        </sup>
      </label>
      <input
        className="input"
        name="phone"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label>
        Organization 
        <sup>
          *
        </sup>
      </label>
      <input
        className="input"
        name="organization"
        onChange={[Function]}
        type="text"
        value=""
      />
      ReCAPCHA not working
      <input
        className="Submit"
        type="submit"
        value="Get This Meal"
      />
    </form>
  </div>
</div>
`;
